src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,371 @@
+  { refine λ x hx, metric.tendsto_at_top.mpr (λ ε hε, _),
+    simp_rw [s, set.compl_Inter, set.compl_Union, set.mem_Union, set.mem_Inter] at hx,
+    obtain ⟨N, hNx⟩ := hx,
+    obtain ⟨k, hk_lt_ε⟩ := h_lt_ε_real ε hε,
+    refine ⟨max N (k - 1), λ n hn_ge, lt_of_le_of_lt _ hk_lt_ε⟩,
+    specialize hNx n ((le_max_left _ _).trans hn_ge),
+    have h_inv_n_le_k : (2 : ℝ)⁻¹ ^ n ≤ 2 * 2⁻¹ ^ k,
+    { rw [mul_comm, ← inv_mul_le_iff' (@two_pos ℝ _ _)],
+      conv_lhs { congr, rw ← pow_one (2 : ℝ)⁻¹ },
+      rw [← pow_add, add_comm],
+      exact pow_le_pow_of_le_one ((one_div (2 : ℝ)) ▸ one_half_pos.le) (inv_le_one one_le_two)
+        ((le_tsub_add.trans (add_le_add_right (le_max_right _ _) 1)).trans
+        (add_le_add_right hn_ge 1)) },
+    refine le_trans _ h_inv_n_le_k,
+    rw [set.mem_compl_iff, set.nmem_set_of_eq, not_le] at hNx,
+    exact hNx.le },
+  rw ae_iff,
+  refine ⟨exists_seq_tendsto_ae.seq_tendsto_ae_seq_strict_mono hfg,
+    measure_mono_null (λ x, _) hμs⟩,
+  rw [set.mem_set_of_eq, ← @not_not (x ∈ s), not_imp_not],
+  exact h_tendsto x,
+end
+
+end exists_seq_tendsto_ae
+
+section tendsto_in_measure_of
+
+variables [measurable_space E] [normed_group E] [borel_space E] [has_measurable_sub₂ E] {p : ℝ≥0∞}
+variables {f : ι → α → E} {g : α → E}
+
+/-- This lemma is superceded by `measure_theory.tendsto_in_measure_of_tendsto_snorm` where we
+allow `p = ∞` and only require `ae_measurable`. -/
+lemma tendsto_in_measure_of_tendsto_snorm_of_measurable
+  (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+  (hf : ∀ n, measurable (f n)) (hg : measurable g) {l : filter ι}
+  (hfg : tendsto (λ n, snorm (f n - g) p μ) l (𝓝 0)) :
+  tendsto_in_measure μ f l g :=
+begin
+  intros ε hε,
+  replace hfg := ennreal.tendsto.const_mul (tendsto.ennrpow_const p.to_real hfg)
+    (or.inr $ @ennreal.of_real_ne_top (1 / ε ^ (p.to_real))),
+  simp only [mul_zero, ennreal.zero_rpow_of_pos (ennreal.to_real_pos hp_ne_zero hp_ne_top)] at hfg,
+  rw ennreal.tendsto_nhds_zero at hfg ⊢,
+  intros δ hδ,
+  refine (hfg δ hδ).mono (λ n hn, _),
+  refine le_trans _ hn,
+  rw [ennreal.of_real_div_of_pos (real.rpow_pos_of_pos hε _), ennreal.of_real_one, mul_comm,
+    mul_one_div, ennreal.le_div_iff_mul_le _ (or.inl (ennreal.of_real_ne_top)), mul_comm],
+  { convert mul_meas_ge_le_pow_snorm' μ hp_ne_zero hp_ne_top ((hf n).sub hg) (ennreal.of_real ε),
+    { exact (ennreal.of_real_rpow_of_pos hε).symm },
+    { ext x,
+      rw [dist_eq_norm, ← ennreal.of_real_le_of_real_iff (norm_nonneg _),
+          of_real_norm_eq_coe_nnnorm] } },
+  { rw [ne, ennreal.of_real_eq_zero, not_le],
+    exact or.inl (real.rpow_pos_of_pos hε _) },
+end
+
+/-- This lemma is superceded by `measure_theory.tendsto_in_measure_of_tendsto_snorm` where we
+allow `p = ∞`. -/
+lemma tendsto_in_measure_of_tendsto_snorm_of_ne_top
+  (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+  (hf : ∀ n, ae_measurable (f n) μ) (hg : ae_measurable g μ) {l : filter ι}
```

Comment:
The assumption `ae_measurable g` should not be necessary here, and in the other lemmas of this section. If `l` is bot, the conclusion of the lemma is trivial, and there is nothing to prove. If `l` is not bot, then you can obtain a sequence `f n` that tends to `g` in `L^p`, then extract a subsequence that converges almost everywhere, and argue that an almost-everywhere limit of ae_measurable functions is ae_measurable. (This is more a fact on L^p spaces than on convergence in measure, by the way).

Reply:
The argument I know for extracting a subsequence which converges ae. from convergence in Lp is that 
convergence in Lp => convergence in measure => subsequence convergent ae.
But since we would like to remove `ae_measurable g` from the first implication this argument is circular. Is there another argument to obtain this subsequence?

Reply:
In fact my idea is wrong, the lemma is not correct without the assumption that `g` is `ae_measurable` (for instance, take a non-measurable function `g` whose norm has integral zero, then it gives a counterexample with `f n = 0` for all `n`).